### ISSUE TITLE
Mobile Responsive

### ISSUE THREAD

**Issue by saileshkush95** [prevwong/craft.js#505]
How can i work for mobile responsive like grapejs in this editor

**Comment by akhrarovsaid** [comment]
I don't believe CraftJS has a styles system that comes out of the box. Fortunately we have some options for the time being. Namely:

1. Build responsiveness into your components. Say you have a container with a class of "container" - well, you can include media queries in your CSS stylesheet that makes that class responsive such that your end user doesn't even have to think about responsive styles. This is less of a "headless" approach but it's pretty straight forward. Then, when your user drags your component, you already have responsiveness figured out for them.

2. Build your own style system. This is a much more daunting but rewarding task. It's difficult, but not impossible. I'd likely look into _Abstract Syntax Trees_ (AST's). The benefit of a CSS AST is that you can parse individual nodes of your CSS and then stringify them into a CSS string which you can include into your pages to have truly dynamic styling options. 

[Here's a Wikipedia page about AST's ](https://en.wikipedia.org/wiki/Abstract_syntax_tree)
Also look into [PostCSS](https://postcss.org/), as it's an excellent AST processor.
Another option is [CSSTree](https://github.com/csstree/csstree).

I hope this helps you!
